File: 01_Array/Q06_189. Rotate Array/solution.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        vector<int> tempArr(nums.size());
        for(int  i = 0; i < nums.size(); i++){
            tempArr[(i+k)%nums.size()] = nums[i];
        }

        nums = tempArr ;
    }
};



// Time Complexity ->O(n)
// Space Complexity -> O(n)

File: 01_Array/Q06_189. Rotate Array/solution.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        n = len(nums)
        tempArr = [0] * n

        for i in range(n):
            tempArr[(i+k) % n ] = nums[i]
        
        nums[:] = tempArr
        
